Assets/Plugins/LeapMotion/Modules/Hands/Scripts/HandBinder/HandBinder.cs

```diff
@@ -126,57 +176,47 @@ public class HandBinder : HandModelBase {
                         }
 
                         //Get the leap bone to extract the position and rotation values
-                        var bone = LeapHand.Fingers[fingerIndex].bones[boneIndex];
+                        var leapBone = LeapHand.Fingers[fingerIndex].bones[boneIndex];
                         //Get any offsets the user has set up
-                        var offset = boundHand.fingers[fingerIndex].boundBones[boneIndex].offset;
+                        var boneOffset = boundHand.fingers[fingerIndex].boundBones[boneIndex].offset;
 
                         //Only update the finger position if the user has defined this behaviour
                         if(SetPositions) {
-                            boundTransform.transform.position = bone.PrevJoint.ToVector3() + offset.position;
+                            boundTransform.transform.position = leapBone.PrevJoint.ToVector3();
+                            boundTransform.transform.localPosition += boneOffset.position;
                         }
+
                         else {
-                            //User can still add offsets to the start position even if they are not setting leap positional data
-                            boundTransform.transform.localPosition = startTransform.position + offset.position;
+                            boundTransform.transform.localPosition = startTransform.position + boneOffset.position;
                         }
 
                         //Update the bound transforms rotation to the leaps rotation * global rotation offset * any further offsets the user has defined
-                        boundTransform.transform.rotation = bone.Rotation.ToQuaternion() * Quaternion.Euler(GlobalFingerRotationOffset) * Quaternion.Euler(offset.rotation);
+                        boundTransform.transform.rotation = leapBone.Rotation.ToQuaternion() * Quaternion.Euler(GlobalFingerRotationOffset) * Quaternion.Euler(boneOffset.rotation);
                     }
                 }
             }
+
+            needsResetting = true;
         }
 
         /// <summary>
         /// Reset the boundGameobjects back to the default pose
         /// </summary>
-        public void ResetHand() {
+        public void ResetHand(bool forceReset = false) {
 
-            //Reset all of the boundTransforms back to position and rotation they were stored to.
-            SetEditorPose = false;
-            foreach(var finger in boundHand.fingers) {
-                if(finger == null) {
-                    continue;
-                }
+            if(defaultHandPose == null || !needsResetting && forceReset != true) {
```

Comment:
Not so keen on lots of negative in a check. Also, we might want to be consistent about checking the boolean state - i.e. !x vs. x==false.